### PR TITLE
bump minio operator version to v4.0.3

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: Xiaoyang Zhu <xiaoyang.zhu@shopee.com>

### CODE REVIEW

File: manifests/samples/harbor_full_minio.yaml [hunk 1]
```diff
@@ -130,7 +130,7 @@ spec:
             - ReadWriteOnce
           resources:
             requests:
-              storage: 10Gi
+              storage: 1Gi
```
Comment (bitsf) [inline manifests/samples/harbor_full_minio.yaml:133]: please also change manifests/samples/harbor_core_minio.yaml

File: pkg/cluster/controllers/storage/tenant.go [hunk 2]
```diff
@@ -96,10 +96,10 @@ func (m *MinIOController) getMinIOProperties(ctx context.Context, harborcluster
 	return storageSpec, nil
 }
 
-func (m *MinIOController) createSecretKeyRef(secretKey []byte, harborcluster *goharborv1.HarborCluster, minioInstance *minio.Tenant) *corev1.Secret {
+func (m *MinIOController) createSecretKeyRef(secretKey []byte, harborcluster *goharborv1.HarborCluster, minioInstance *miniov2.Tenant) *corev1.Secret {
 	s3KeySecret := &corev1.Secret{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: "v1",
+			APIVersion: "minio.min.io",
```
Comment (steven-zou) [inline pkg/cluster/controllers/storage/tenant.go:102]: Not `minio.min.io/v2`?

**Review by steven-zou — COMMENTED** [review]
Please check my comments.

File: pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go [hunk 3]
```diff
@@ -0,0 +1,926 @@
...
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced,shortName=tenant,singular=tenant
+// +kubebuilder:printcolumn:name="State",type="string",JSONPath=".status.currentState"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:storageversion
+
```
Comment (heww) [inline pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go:17]: These comments need to be removed and add `// +kubebuilder:skip` in this file.

File: pkg/cluster/controllers/storage/minio.go [hunk 4]
```diff
@@ -124,7 +125,7 @@ func (m *MinIOController) minioInit(ctx context.Context, harborcluster *goharbor
 		return err
 	}
 
-	endpoint := m.getServiceName(harborcluster) + "." + harborcluster.Namespace + ":9000"
+	endpoint := m.getTenantsServiceName(harborcluster) + "." + harborcluster.Namespace + ":" + m.getServicePort()
```
[the remark this answers is not in the document]
Comment (bitsf) [reply pkg/cluster/controllers/storage/minio.go:128]: minio can also support https, I think it should not be only 80

File: pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go (hunk 3 above)
Comment (holyhope) [reply pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go:17]: Why `// +kubebuilder:skip`? Having crd matching these structure is not needed? Then how to create/update these resources?

File: pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go [hunk 5]
```diff
@@ -0,0 +1,926 @@
+package v2
```
Comment (holyhope) [inline pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go:1]: why having apis files somewhere else than in the top level package `/apis`?

File: pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go [hunk 6]
```diff
@@ -0,0 +1,926 @@
...
+	Image string `json:"image,omitempty"`
+	// *Optional* +
+	//
+	// Deprecated: in Operator v4.0.1* +
+	//
+	// Defines the Docker image to use as a sidecar for the Prometheus server. Defaults to `alpine`. +
+	//
+	// The specified Docker image *must* be the https://hub.docker.com/_/alpine[`alpine`] package. +
+	// +optional
+	SideCarImage string `json:"sidecarimage,omitempty"`
```
Comment (holyhope) [inline pkg/cluster/controllers/storage/minio/apis/minio.min.io/v2/types.go:666]: Not sure why this structure is under `.../storage/minio/.../types.go`. I suggest to split this `types.go` file to multiple smaller one.
WDYT?

File: pkg/cluster/controllers/storage/tenant.go (hunk 2 above)
Comment (holyhope) [reply pkg/cluster/controllers/storage/tenant.go:102]: Are you sure that the `corev1.Secret` should use the APIVersion `minio.min.io`? IMO there is some confusion here.
Once the [scheme is added to the runtime](https://github.com/goharbor/harbor-operator/blob/master/pkg/scheme/scheme.go). The `TypeMeta` field does not need to be specified.

**Pull request closed without merging**